include/magma/Deleter.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "magma/Device.hpp"
+
+namespace magma
+{
+  template<class T>
```

Comment:
Rather than template the class, I would template it's `operator()`.
I don't think having a templated class brings us anything here.